builder/openstack_id3/access_config.go

```diff
@@ -0,0 +1,150 @@
+	DomainId  string `mapstructure:"domain_id"`
+}
+
+// Auth returns a ProviderClient with an valid token and catalog for access to openstack services,
+// or an error if the authentication couldn't be resolved.
+func (c *AccessConfig) Auth() (*gophercloud.ProviderClient, error) {
+	// Note: Some of the fetched env vars here are a bit odd but needs to be here because of compatibility
+	c.Username = common.ChooseString(c.Username, os.Getenv("SDK_USERNAME"), os.Getenv("OS_USERNAME"))
+	c.UserId = common.ChooseString(c.UserId, os.Getenv("OS_USER_ID"))
+	c.Password = common.ChooseString(c.Password, os.Getenv("SDK_PASSWORD"), os.Getenv("OS_PASSWORD"))
+	c.ApiKey = common.ChooseString(c.ApiKey, os.Getenv("SDK_API_KEY"))
+	c.Project = common.ChooseString(c.Project, os.Getenv("SDK_PROJECT"), os.Getenv("OS_TENANT_NAME"), os.Getenv("OS_PROJECT_NAME"))
+	c.Provider = common.ChooseString(c.Provider, os.Getenv("SDK_PROVIDER"), os.Getenv("OS_AUTH_URL"))
+	c.RawRegion = common.ChooseString(c.RawRegion, os.Getenv("SDK_REGION"), os.Getenv("OS_REGION_NAME"))
+	c.TenantId = common.ChooseString(c.TenantId, c.ProjectId, os.Getenv("OS_TENANT_ID"), os.Getenv("OS_PROJECT_ID"))
+	c.Domain = common.ChooseString(c.Domain, os.Getenv("OS_DOMAIN_NAME"))
+	c.DomainId = common.ChooseString(c.DomainId, os.Getenv("OS_PROJECT_DOMAIN_ID"), os.Getenv("OS_USER_DOMAIN_ID"))
+
+	authoptions := gophercloud.AuthOptions{
+		AllowReauth: true,
+
+		IdentityEndpoint: c.Provider,
+		Username:         c.Username,
+		UserID:           c.UserId,
+		Password:         c.Password,
+		APIKey:           c.ApiKey,
+		TenantName:       c.Project,
+		TenantID:         c.TenantId,
+		DomainName:       c.Domain,
+		DomainID:         c.DomainId,
+	}
+
+	default_transport := &http.Transport{}
+
+	if c.Insecure {
+		cfg := new(tls.Config)
+		cfg.InsecureSkipVerify = true
+		default_transport.TLSClientConfig = cfg
+	}
+
+	// For corporate networks it may be the case where we want our API calls
+	// to be sent through a separate HTTP proxy than external traffic.
+	if c.ProxyUrl != "" {
+		url, err := url.Parse(c.ProxyUrl)
+		if err != nil {
+			return nil, err
+		}
+
+		// The gophercloud.Context has a UseCustomClient method which
+		// would allow us to override with a new instance of http.Client.
+		default_transport.Proxy = http.ProxyURL(url)
+	}
+
+	if c.Insecure || c.ProxyUrl != "" {
+		http.DefaultTransport = default_transport
+	}
+
+	// Creates the provider client that will contain token and catalog
+	provider, err := openstack.AuthenticatedClient(authoptions)
+	if err != nil {
+		return nil, err
+	}
+	// Attempts to autheticate and update provider with token and catalog
+	err = openstack.Authenticate(provider, authoptions)
+	if err != nil {
+		return nil, err
+	}
+	return provider, nil
+}
+
+func (c *AccessConfig) Region() string {
+	return common.ChooseString(c.RawRegion, os.Getenv("SDK_REGION"), os.Getenv("OS_REGION_NAME"))
+}
+
+func (c *AccessConfig) Prepare(t *packer.ConfigTemplate) []error {
```

Comment:
I think this should probably be a [multierror](https://github.com/hashicorp/go-multierror). What do you think?